doc/source/ray-observability/state/state-api.rst

```diff
@@ -134,143 +177,331 @@ It is recommended to start monitoring states through summary APIs first. When yo
 By default, the API prints log from a head node.
 
-E.g., Get all retrievable log file names
+E.g., Get all retrievable log file names from a head node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabbed:: CLI
 
-.. code-block:: bash
+    .. code-block:: bash
 
-    ray logs 
+        ray logs 
+
+.. tabbed:: Python SDK
+
+    .. code-block:: python
+
+        # You could get the node id / node ip from `ray list nodes` 
+        from ray.experimental.state.api import list_logs 
+        # `ray logs` by default print logs from a head node. 
+        # So in order to list the same logs, you should provide the head node id. 
+        # You could get the node id / node ip from `ray list nodes` 
+        list_logs(node_id=<HEAD_NODE_ID>)
 
 E.g., Get a particular log file from a node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabbed:: CLI
+
+    .. code-block:: bash
+
+        # You could get the node id / node ip from `ray list nodes` 
+        ray logs gcs_server.out --node-id <NODE_ID> 
 
-.. code-block:: bash
+.. tabbed:: Python SDK
 
-    # You could get the node id / node ip from `ray list nodes` 
-    ray logs gcs_server.out --node-id <XYZ> 
+    .. code-block:: python
+
+        from ray.experimental.state.api import get_log 
+
+        # Node IP could be retrieved from list_nodes() or ray.nodes()
+        for line in get_log(filename="gcs_server.out", node_id=<NODE_ID>):
+            print(line)
 
 E.g., Stream a log file from a node
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabbed:: CLI
 
-.. code-block:: bash
+    .. code-block:: bash
 
-    ray logs -f raylet.out --node-ip 172.31.47.143
+        # You could get the node id / node ip from `ray list nodes` 
+        ray logs -f raylet.out --node-ip <NODE_IP> 
 
-E.g., Stream actor log with actor id 
+.. tabbed:: Python SDK
 
-.. code-block:: bash
+    .. code-block:: python
+
+        from ray.experimental.state.api import get_log 
+
+        # Node IP could be retrieved from list_nodes() or ray.nodes()
+        for line in get_log(filename="raylet.out", node_ip=<NODE_IP>, follow=True):
+            print(line)
 
-    # You could use ray list actors to get the actor ids
-    ray logs --actor-id=<XXX>
 
 E.g., Stream log from a pid 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabbed:: CLI
+
+    .. code-block:: bash
+
+        ray logs --pid=<PID> --follow
+
+.. tabbed:: Python SDK
+
+    .. code-block:: python
 
-.. code-block:: bash
+        from ray.experimental.state.api import get_log 
 
-    ray logs --pid=<XXX> --follow
+        # Node IP could be retrieved from list_nodes() or ray.nodes()
+        # You could get the pid of the worker running the actor easily when output
+        # of worker being directed to the driver (default)
+        # The loop will block with `follow=True`
+        for line in get_log(pid=<PID>, node_ip=<NODE_IP>, follow=True):
```

Comment:
This will no longer be relevant after the logging refactor (which allows auto-population of `node_ip` and `node_id`). See [this](https://github.com/ray-project/ray/pull/27069/files#diff-1f9ff1a83a3da4525dd3be04b6aa2fb29e862e6484f46e6676ce38652373291bR1131)